src/pydap/net.py

```diff
@@ -25,6 +29,10 @@ def GET(url, application=None, session=None, timeout=DEFAULT_TIMEOUT,
         _, _, path, query, fragment = urlsplit(url)
         url = urlunsplit(('', '', path, query, fragment))
 
+    scheme, host, path, query, fragment = urlsplit(url)
+    query = quote(query)
+    url = urlunsplit((scheme, host, path, query, fragment))
```

Comment:
I think there should be a unit test or two added to ensure the URL query gets encoded as expected. The things that I think would make a good test:
* Seeing a URL with the `[` and `]` characters in the query (so I'm guessing a constraint expression).
* Making sure slashes in variable names are handled correctly (I _think_ `quote` may consider `/` to be a safe character).
* Making sure the rest of the URL is not encoded.

I think one way to do this test would potentially be to make the `follow_redirect` function and then look at the value of the `url` argument.